src/Catalog.API/Apis/CatalogApi.cs

```diff
@@ -8,12 +9,26 @@ namespace eShop.Catalog.API;
 
 public static class CatalogApi
 {
-    public static IEndpointRouteBuilder MapCatalogApiV1(this IEndpointRouteBuilder app)
+    public static IEndpointRouteBuilder MapCatalogApi(this IEndpointRouteBuilder app)
     {
-        var api = app.MapGroup("api/catalog").HasApiVersion(1.0);
+        // RouteGroupBuilder for catalog endpoints
+        // var vapi = app.NewVersionedApi("Catalog").MapGroup("api/catalog");
+        // var api = vapi.HasApiVersion(1.0).HasApiVersion(2.0);
+        // var v1 = vapi.HasApiVersion(1.0);
+        // var v2 = vapi.HasApiVersion(2.0);
+
+        var vApi = app.NewVersionedApi("Catalog");
+        var api = vApi.MapGroup("api/catalog").HasApiVersion(1, 0).HasApiVersion(2, 0);
+        var v1 = vApi.MapGroup("api/catalog").HasApiVersion(1, 0);
+        var v2 = vApi.MapGroup("api/catalog").HasApiVersion(2, 0);
 
         // Routes for querying catalog items.
-        api.MapGet("/items", GetAllItems)
+        v1.MapGet("/items", GetAllItemsV1)
+            // .WithName("ListItems")
```

Comment:
Maybe we can try the following for now?

```suggestion
            .WithName("ListItems-v1")
```

I think in an ideal world, there is some sort of `WithVersionedName` extension made that will generate a globally unique endpoint name by prepending the API version somewhere but this might be a good stop-gap.